### PR TITLE
Configurable parameter simplification

### PR DESCRIPTION
**Pull request by creisle**
Improvements
- Removed any unnecessary parameters
- checker is now detecting command not found errors
- filters spanning read alignments using the min contig score parameter (same filtering done on contigs)

New features
- Added mail options for slurm/sge

BugFixes
- Filtering contig alignments was getting highest rank instead of lowest. Best ranks are lower ranks
- logs version for the configured aligner instead of just blat

### CODE REVIEW

File: mavis/assemble.py [hunk 1]
```diff
@@ -311,70 +311,54 @@ def filter_contigs(contigs, assembly_min_uniq=0.01):
...
+    drops any sequences too small to fit the kmer size
+
     Args:
         sequences (:class:`list` of :class:`str`): a list of strings/sequences to assemble
-        assembly_max_kmer_size: see :term:`assembly_max_kmer_size`
-        assembly_min_nc_edge_weight: see :term:`assembly_min_nc_edge_weight`
-        assembly_min_edge_weight: see :term:`assembly_min_edge_weight`
+        kmer_size: see :term:`assembly_kmer_size` the size of the kmer to use
+        min_edge_trim_weight: see :term:`min_edge_trim_weight`
+        min_edge_trim_weight: see :term:`min_edge_trim_weight`
```
Comment (calchoo) [inline mavis/assemble.py]: We can remove one of these lines

File: mavis/bam/read.py [hunk 2]
```diff
@@ -171,7 +171,7 @@ def get_samtools_version():
             mid = int(match.group('mid')) if match.group('mid') else 0
             minor = int(match.group('minor')) if match.group('minor') else 0
             return major, mid, minor
-    raise ValueError('unable to parse samtools version number')
+    raise ValueError('unable to parse samtools version number', proc)
```
Comment (calchoo) [inline mavis/bam/read.py]: Should we make the string the same as the aligner for consistency?
 + raise ValueError("unable to parse blat version number from:'{}'".format(proc))

Comment (creisle) [reply mavis/bam/read.py]: but this is samtools and not the aligner

Comment (creisle) [reply mavis/bam/read.py]: sure I will add this in the other branch

**Pull request merged**